src/researches/findKeywordInPageTitle.js

```diff
@@ -3,7 +3,35 @@
 import wordMatch from "../stringProcessing/matchTextWithWord.js";
 const findTopicFormsInString = require( "./findKeywordFormsInString.js" ).findTopicFormsInString;
 
-import { escapeRegExp, includes } from "lodash-es";
+import getFunctionWordsFactory from "../helpers/getFunctionWords";
+
+import { escapeRegExp, get, includes, isUndefined } from "lodash-es";
+import getLanguage from "../helpers/getLanguage";
+import addWordboundary from "../stringProcessing/addWordboundary";
+
+const getFunctionWords = getFunctionWordsFactory();
+
+/**
+ * Strips all function words from the start of the given string.
+ * @param {string[]} functionWords The array of function words to strip from the start.
+ * @param {string} str The string from which to strip the function words.
+ * @returns {string} The given string with the function words stripped.
+ */
+const stripFunctionWordsFromStart = function( functionWords, str ) {
```

Comment:
This function now goes through every function word in the list (e.g., almost 3000 words for Russian), turns it into a regex and then tries to find them back in the title string. I think it is much more efficient to go the other way around. If an exact match of the keyphrase was already found in the title, get words from the title and check if any (eventual) words before the exact match belong to the function words list.

Reply:
Nice catch! This will make it even more efficient 💪 .